test/framework/cluster.go

```diff
@@ -661,12 +661,36 @@ func (e *ClusterE2ETest) GetCapiMachinesForCluster(clusterName string) map[strin
 	return machinesMap
 }
 
+// ApplyClusterManifest uses client-side logic to create/update objects defined in a cluster yaml manifest.
+func (e *ClusterE2ETest) ApplyClusterManifest() {
+	ctx := context.Background()
+	e.T.Logf("Applying cluster %s spec located at %s", e.ClusterName, e.ClusterConfigLocation)
+	e.applyClusterManifest(ctx)
+	e.ValidateClusterState()
+	e.StopIfFailed()
+}
+
+func (e *ClusterE2ETest) applyClusterManifest(ctx context.Context) {
+	if err := e.KubectlClient.ApplyManifest(ctx, e.kubeconfigFilePath(), e.ClusterConfigLocation); err != nil {
+		e.T.Fatalf("Failed to apply cluster config: %s", err)
+	}
+}
+
 func WithClusterUpgrade(fillers ...api.ClusterFiller) ClusterE2ETestOpt {
 	return func(e *ClusterE2ETest) {
 		e.UpdateClusterConfig(api.ClusterToConfigFiller(fillers...))
 	}
 }
 
+// UpgradeClusterWithKubectl uses client-side logic to upgrade a cluster.
+func (e *ClusterE2ETest) UpgradeClusterWithKubectl(clusterOpts []ClusterE2ETestOpt, commandOpts ...CommandOpt) {
+	for _, opt := range clusterOpts {
+		opt(e)
+	}
+	e.buildClusterConfigFile()
+	e.ApplyClusterManifest()
+}
```

Comment:
Can we add this method to `/test/framework/workload.go` and use the `ApplyClusterManifest` instead there as well?